core/modules/catalog/index.ts

```diff
@@ -17,4 +22,113 @@ export const CatalogModule: StorefrontModule = function (app, store, router, mod
+          {
+            [filterOption.attribute_code]: prevOption
+          }
+        ))
+      } else {
+        store.commit(PRODUCT_SET_CURRENT_CONFIGURATION, Object.assign(
+          {},
+          store.getters['product/getCurrentProductConfiguration'],
+          {
+            [filterOption.attribute_code]: undefined
+          }
+        ))
+      }
+    }
+  })
+
+  EventBus.$on('product-after-customoptions', async (payload) => {
+    let priceDelta = 0
+    let priceDeltaInclTax = 0
+    const optionValues: any[] = Object.values(payload.optionValues)
+    optionValues.forEach(optionValue => {
+      if (optionValue && parseInt(optionValue.option_type_id) > 0) {
+        if (optionValue.price_type === 'fixed' && optionValue.price !== 0) {
+          priceDelta += optionValue.price
+          priceDeltaInclTax += optionValue.price
+        }
+        if (optionValue.price_type === 'percent' && optionValue.price !== 0) {
+          priceDelta += ((optionValue.price / 100) * store.getters['product/getOriginalProduct'].price)
+          priceDeltaInclTax += ((optionValue.price / 100) * store.getters['product/getOriginalProduct'].price_incl_tax)
+        }
+      }
+    })
+    store.commit(PRODUCT_SET_CURRENT, Object.assign(
+      {},
+      store.getters['product/getCurrentProduct'],
+      {
+        price: store.getters['product/getOriginalProduct'].price + priceDelta,
+        price_incl_tax: store.getters['product/getOriginalProduct'].price_incl_tax + priceDeltaInclTax
+      }
+    ))
+  })
+
+  EventBus.$on('product-after-bundleoptions', async (payload) => {
+    let priceDelta = 0
+    let priceDeltaInclTax = 0
+    const optionValues: any[] = Object.values(payload.optionValues)
+    optionValues.forEach(optionValue => {
+      if (optionValue && optionValue.value && optionValue.product && parseInt(optionValue.qty) >= 0) {
+        priceDelta += optionValue.value.product.price * parseInt(optionValue.qty)
+        priceDeltaInclTax += optionValue.value.product.price_incl_tax * parseInt(optionValue.qty)
+      }
+    })
+    if (priceDelta > 0) {
+      store.commit(PRODUCT_SET_CURRENT, Object.assign(
+        {},
+        store.getters['product/getCurrentProduct'],
+        {
+          price: priceDelta,
+          price_incl_tax: priceDeltaInclTax
+        }
+      ))
+    }
+  })
+
+  const onUserPricesRefreshed = async () => {
+    if (router.currentRoute.params.parentSku) {
+      await store.dispatch('product/reset')
+      await store.dispatch('product/single', {
+        options: {
+          sku: router.currentRoute.params.parentSku,
+          childSku: router && router.currentRoute.params && router.currentRoute.params.childSku ? router.currentRoute.params.childSku : null
+        },
+        skipCache: true
+      })
+    }
+  }
+
+  if (config.usePriceTiers || isUserGroupedTaxActive()) {
+    EventBus.$on('user-after-loggedin', onUserPricesRefreshed)
+    EventBus.$on('user-after-logout', onUserPricesRefreshed)
+  }
```

Comment:
The only suggestion to changes in this file is that maybe we could split these event handlers into separate helpers so they'd be easier to test / re-use?

Reply:
i think this should be next step - get all there Listeners in this file, then find when events are emitted and change them to actions to to this job. WDYT?

Reply:
Yeah, sounds OK but at this point, we can just move the functions (handlers) into separate helpers - that's all, to keep this index.ts smaller and more readable